### ISSUE TITLE
Questions about Figure 4, GRPO loss/KL relationship, and reproducibility

### ISSUE THREAD

**Issue by zhangp365** [Leke-G/UAV-VL-R1#4]
Hi authors,

Thank you for releasing the paper **“UAV-VL-R1: Generalizing Vision-Language Models via Supervised Fine-Tuning and Multi-Stage GRPO for UAV Visual Reasoning.”**

While reading the paper, I noticed several points regarding Figure 4 and the GRPO implementation that I could not reconcile. I would appreciate clarification and, if possible, the corresponding training logs/configurations.

### 1. The repository currently contains no released source code

The paper states that the source code and datasets are available in this GitHub repository. However, at the moment, the repository appears to contain only a short README:

> “This repository stores the source code of the upcoming published paper experiments. Please waiting”

I could not find:

* GRPO training code
* training configurations
* reward implementation
* KL coefficient (`beta`)
* training logs
* checkpoints
* dataset files/scripts
* Figure 4 plotting code or raw data

Could you please clarify when the code and experimental artifacts will be released?

### 2. Loss and KL divergence in Figure 4 appear almost exactly proportional

In Figure 4, the loss curve and KL-divergence curve are plotted on separate y-axes and appear not merely correlated, but almost perfectly aligned, including local peaks and fluctuations.

From the axis scales, the approximate relationship appears to be:

```text
Loss ≈ 0.04 × KL
```

For example:

```text
KL ≈ 0.30  -> Loss ≈ 0.012
KL ≈ 0.15  -> Loss ≈ 0.006
KL ≈ 0.10  -> Loss ≈ 0.004
```

Since the GRPO objective contains both the policy/advantage term and the KL regularization term, I would normally expect something conceptually similar to:

```text
total_loss = policy_loss + beta * KL
```

(up to sign conventions and implementation details).

Therefore, the near-perfect proportionality raises a few questions:

1. What exact value of `beta` was used?
2. What exactly does the plotted “Loss” represent?
3. Is it the complete GRPO loss, or only the KL-weighted component?
4. What is the magnitude of the policy/surrogate loss relative to `beta * KL`?
5. Were both curves generated directly from independent logged metrics?

It would be very helpful if you could provide the raw values for:

```text
step
total_loss
policy_loss
KL
beta * KL
reward
format_reward
accuracy_reward
```

### 3. The textual description of Figure 4 seems inconsistent with the plotted curve

The paper states:

> “the training loss steadily decreases and begins to stabilize around step 1400”

However, visually, the loss seems to:

* increase substantially during the early/middle stages,
* reach a pronounced peak around approximately step 800,
* drop sharply afterward,
* rise again,
* and continue fluctuating later in training.

Therefore, “steadily decreases” does not seem to describe the plotted trajectory accurately.

Could you clarify whether:

* the wrong curve was plotted,
* the description corresponds to another run,
* smoothing was applied,
* or the figure was generated from different logs than those discussed in the text?

### 4. Possible inconsistency in the GRPO advantage equation

Equation (4) appears to define the normalized advantage as:

```text
A_i = (r_i - max({r_1, ..., r_G})) / std({r_1, ..., r_G})
```

However, the surrounding text discusses the **mean reward**, and standard GRPO-style group normalization would normally use something of the form:

```text
A_i = (r_i - mean(r)) / std(r)
```

If `max(r)` is actually used, then:

```text
A_i <= 0
```

for every sampled response, with the maximum-reward response having exactly zero advantage.

Could you confirm whether `max` in Eq. (4) is a typo and the implementation actually uses `mean`?

### 5. Request for minimal reproducibility artifacts

To make Figure 4 independently verifiable, even a small CSV/log file would be sufficient, for example:

```text
step,total_loss,policy_loss,kl,format_reward,accuracy_reward
```

together with the relevant GRPO hyperparameters:

```text
beta
epsilon / clip range
num_generations / group size
learning_rate
temperature
batch size
gradient accumulation
```

I am not claiming that the reported results are incorrect. I am asking because, with the currently released repository, Figure 4 and the GRPO dynamics cannot be independently reproduced or checked, and the almost exact alignment between Loss and KL is unusual enough to deserve clarification.

Thank you.
